lib/processor.js

```diff
@@ -145,6 +156,24 @@ function preprocess(text) {
  * @returns {Message[]} A flattened array of messages with mapped locations.
  */
 function postprocess(messages) {
+
+    /**
+     * Returns messages which are not in the list of expected errors
+     * @param  {Message[]}  msgs A flattened array of messages with mapped locations.
+     * @returns {Message[]}      An array of messages without expected messages.
+     */
+    function getUnexpectedMessages(msgs) {
+        return msgs.filter(function (msg) {
+            var msgIsExpected = false;
+            expectedErrors.forEach(function (expErr) {
```

Comment:
Map as in hash table in this case. Right now, for each error message, it iterates linearly through the list of expected errors to find one where the line number matches. If there are m errors and n expected, the big-O notation (warning: CS theory here) for that running time would be O(m*n). If instead there were a table mapping line number to expected errors (if any), then it would be possible to ask immediately "are there any expected errors on line x?", skipping the cost of iterating the expected errors list. This check would look something like `if (expectedErrors[error.line] === error.message)`, with `expectedErrors` being either a JS object with integer keys or a sparse array. This is the same thing I did in the TokenStore, creating a map from integer range indices to tokens, so we could look up tokens directly by their starting locations ("give me the token starting at range index 32", or "find the first token of this BinaryExpression by their shared starting location").

Hopefully this isn't too pedantic, feel free to skim the parts you already know.